Source/DafnyCore/AST/Expressions.cs

```diff
@@ -3744,7 +3690,8 @@ public class NestedMatchCaseStmt : NestedMatchCase, IAttributeBearingDeclaration
     this.Attributes = attrs;
   }
 
-  public override IEnumerable<INode> Children => Body.Concat<INode>(Attributes?.Args ?? Enumerable.Empty<INode>());
+  public override IEnumerable<INode> Children =>
+    (Attributes != null ? new List<INode> { Attributes } : Enumerable.Empty<INode>()).Concat(Body).Concat<INode>(Attributes?.Args ?? Enumerable.Empty<INode>());
```

Comment:
Shouldn't `Attributes.Children` return `Args` _and_ `Prev` ?

Reply:
Yes it does, but that would seems wrong from a modelisation's perspective that the first attribute is flattend in the INode tree, and not the second.

Reply:
I think there's something I'm missing. It looks to me as if given the current definition of `Attributes.Children`, you should remove `.Concat<INode>(Attributes?.Args ?? Enumerable.Empty<INode>())` here to prevent getting the arguments of the first `Attributes` as descendants of `NestedMatchCaseStmt` twice.

Reply:
Good catch. This was the previous code. I removed it, as it was even out of order with the `Body` node